browser/ipfs/ipfs_host_resolver.cc

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+#include <vector>
+
+#include "brave/browser/ipfs/ipfs_host_resolver.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "net/base/host_port_pair.h"
+#include "net/dns/public/dns_protocol.h"
+
+namespace ipfs {
+
+IPFSHostResolver::IPFSHostResolver(
+    network::mojom::NetworkContext* network_context,
+    const std::string& prefix)
+    : prefix_(prefix), network_context_(network_context) {
+  DCHECK(network_context);
+}
+IPFSHostResolver::~IPFSHostResolver() {}
+
+void IPFSHostResolver::Resolve(const net::HostPortPair& host,
+                               const net::NetworkIsolationKey& isolation_key,
+                               net::DnsQueryType dns_query_type,
+                               HostTextResultsCallback callback) {
+  if (!callback)
+    return;
+  if (IsRunning())
+    Reset();
+
+  resolving_host_ = host.host();
+  net::HostPortPair local_host_port(prefix_ + resolving_host_, host.port());
+
+  network::mojom::ResolveHostParametersPtr parameters =
+      network::mojom::ResolveHostParameters::New();
+  parameters->dns_query_type = dns_query_type;
+
+  receiver_.reset();
+  resolved_callback_ = std::move(callback);
+
+  network_context_->ResolveHost(local_host_port, isolation_key,
+                                std::move(parameters),
+                                receiver_.BindNewPipeAndPassRemote());
+  receiver_.set_disconnect_handler(base::BindOnce(
+      &IPFSHostResolver::OnMojoConnectionError, base::Unretained(this)));
+}
+
+bool IPFSHostResolver::IsRunning() const {
+  return !resolving_host_.empty();
+}
+
+void IPFSHostResolver::Reset() {
+  resolving_host_.erase();
+  resolved_callback_.Reset();
+
+  if (reset_callback_for_testing_)
+    std::move(reset_callback_for_testing_).Run();
+}
+
+void IPFSHostResolver::OnComplete(
+    int result,
+    const net::ResolveErrorInfo& error_info,
+    const base::Optional<net::AddressList>& list) {
+  if (result != net::OK) {
+    VLOG(1) << "DNS resolving error:" << net::ErrorToString(result)
+            << " for host: " << prefix_ + resolving_host_;
+  }
+  Reset();
+}
+
+void IPFSHostResolver::OnTextResults(const std::vector<std::string>& results) {
+  VLOG(1) << results.size()
```

Comment:
minor: this should probably be at a higher logging level since it's not an error

Reply:
done